pandas/core/groupby/ops.py

```diff
@@ -575,6 +565,11 @@ def _cython_operation(
         elif is_datetimelike and kind == "aggregate":
             result = result.astype(orig_values.dtype)
 
+        if is_integer_dtype(orig_values.dtype) and is_extension_array_dtype(
```

Comment:
the whole entire point of `maybe_cast_result` is that you don't need the if here

to be honest we should remove lines 558-566, but can do that in another PR

Reply:
Without the if about 550 groupby tests fail, though I think it works when only checking for extension arrays